### PR TITLE
Start migrating "MapConfigService" to State Tree

### PR DESCRIPTION
**Pull request by theduckylittle**
The MapConfigService needs to be deprecated and replaced with a dedicated part of the state tree which holds the critical map information.

This is a start on that in order to support some of the work being done by @CarlyLanglois .

### CODE REVIEW

File: src/reducers/map.js [hunk 1]
```diff
@@ -20,17 +20,52 @@ function convertFromExtent(mapSize, extent, rotation) {
...
+  const p = (new ol.geom.Point(pt)).transform(srcProj, destProj);
+  return p.getCoordinates();
+}
+
+const defaultState = {
+  config: {
+    // default projection for OpenLayers maps.
+    projection: 'EPSG:3857',
+    // bounds are specified in EPSG:4326 for interoperability.
+    bounds: [-180.0, -90.0, 180.0, 90.0]
```
Comment (bartvde) [inline src/reducers/map.js]: are we doing bounds or center and zoom?

File: src/reducers/map.js [hunk 2]
```diff
@@ -20,17 +20,52 @@ function convertFromExtent(mapSize, extent, rotation) {
...
+        ...state,
+        config: {
+          ...state.config,
+          bounds: action.bounds
+        }
       }
     case  MAP.SET_VIEW:
       const new_view = state.view;
+
+      // if the action specifys a projection when setting the view,
```
Comment (bartvde) [inline src/reducers/map.js]: specifies

Comment (theduckylittle) [inline src/reducers/map.js]: Doh.

File: src/reducers/map.js (hunk 1 above)
Comment (theduckylittle) [inline src/reducers/map.js]: Changed to maxExtent for clarity.  We're using center/resolution to determine the "view", this is meant to define the map's overall configuration and I was just stubbing in the ability to limit the map's maximum bounds.

**Comment by theduckylittle** [comment]
With @bartvde 's approval, i'm merging this in since we're using the improvements in other components.

**Pull request merged**